app/ui/Project/ProjectModal.tsx

```diff
@@ -0,0 +1,192 @@
+"use client";
+import { generateRandomNumber } from "@/app/lib/utils";
+import { addProject, selectProject } from "@/app/redux/projectSlice";
+import { useAppSelector } from "@/app/redux/store";
+import { setTask } from "@/app/redux/taskSlice";
+import { ProjectData, ProjectModalProps } from "@/definition";
+import { Button, Modal } from "flowbite-react";
+import { redirect } from "next/navigation";
+import { useEffect, useRef, useState } from "react";
+import { HiDotsHorizontal } from "react-icons/hi";
+import { PiWarningDiamondFill } from "react-icons/pi";
+import { useDispatch } from "react-redux";
+
+const ProjectModal: React.FC<ProjectModalProps> = ({ email }) => {
+  const [openModal, setOpenModal] = useState(false);
+  const projectRef = useRef<HTMLInputElement | null>(null);
+  const descriptionRef = useRef<HTMLTextAreaElement | null>(null);
+  const dateRef = useRef<HTMLInputElement | null>(null);
+  const emailsRef = useRef<HTMLInputElement | null>(null);
+  const [teamMember, setTeamMember] = useState([]);
+  const [error, setError] = useState<string | null>(null);
+  const dispatch = useDispatch();
+  const Project = useAppSelector(selectProject);
+
+  useEffect(() => {
+    if (
+      Project.title &&
+      Project.id &&
+      Project.description &&
+      Project.createdBy &&
+      Project.date
+    ) {
+      redirect("/project/boards");
+    }
+  }, [Project]);
+  useEffect(() => {
+    console.log(teamMember);
+  }, [teamMember]);
+
+  const CreateProject = async () => {
+    if (
+      projectRef.current?.value === "" ||
+      descriptionRef.current?.value === "" ||
+      dateRef.current?.value === ""
+    ) {
+      setError("Please fill all fields");
+      return;
+    }
+    getEmails();
+
+    const project: ProjectData = {
+      id: generateRandomNumber(),
+      title: projectRef.current?.value || "",
+      description: descriptionRef.current?.value || "",
+      date: dateRef.current?.value
+        ? new Date(dateRef.current.value).toLocaleDateString()
+        : new Date().toLocaleDateString(),
+      createdBy: email,
+      team: teamMember,
+    };
+    await dispatch(addProject(project));
+    await dispatch(setTask(project.id));
+  };
+
+  const closeModal = () => {
+    setOpenModal(false);
+    setError(null);
+    if (projectRef.current) projectRef.current.value = "";
+  };
+
+  const getEmails = () => {
+    const emailInput = emailsRef.current;
+    const emailList = emailInput?.value.split(",") || [];
+
+    if (emailList?.length > 0) {
+      emailList?.forEach((email) => {
+        email = email.trim();
+
+        const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+        const isValid = emailPattern.test(email);
+
```

Comment:
put email validation in utils file so that you can use email validation at any place from there